client/src/elm/Pages/Prenatal/Activity/Utils.elm

```diff
@@ -1192,14 +1359,187 @@ healthEducationFormFamilyPlanningInput language isChw form =
+                        (SetHealthEducationSubActivityBoolInput (\value form_ -> { form_ | constipation = Just value }))
+                        "constipation"
+                        Nothing
+                  ]
+                , Just form.constipation
+                )
+
+            else
+                ( [], Nothing )
+
+        heartburn =
+            let
+                reliefMethods =
+                    List.map
+                        (Translate.HeartburnReliefMethod
+                            >> translate language
+                            >> String.toLower
+                            >> text
+                            >> List.singleton
+                            >> li []
+                        )
+                        [ ReliefMethodAvoidLargeMeals
+                        , ReliefMethodCeaseSmoking
+                        , ReliefMethodAvoidAlcohom
+                        , ReliefMethodSleepWithHeadRaised
+                        ]
+                        |> ul []
+            in
+            if diagnosed DiagnosisHeartburn assembled then
+                ( [ viewCustomLabel language (Translate.PrenatalHealthEducationLabel EducationHeartburn) "" "label header"
+                  , viewLabel language Translate.PrenatalHealthEducationHeartburnInform
+                  , reliefMethods
+                  , viewQuestionLabel language Translate.PrenatalHealthEducationAppropriateProvided
+                  , viewBoolInput
+                        language
+                        form.heartburn
+                        (SetHealthEducationSubActivityBoolInput (\value form_ -> { form_ | heartburn = Just value }))
+                        "heartburn"
+                        Nothing
+                  ]
+                , Just form.heartburn
+                )
 
+            else
+                ( [], Nothing )
+
+        varicoseVeins =
+            if symptomRecorded assembled.measurements VaricoseVeins then
+                ( [ viewCustomLabel language (Translate.PrenatalHealthEducationLabel EducationVaricoseVeins) "" "label header"
+                  , viewCustomLabel language Translate.PrenatalHealthEducationVaricoseVeinsInform "." "label paragraph"
+                  , viewQuestionLabel language Translate.PrenatalHealthEducationAppropriateProvided
+                  , viewBoolInput
+                        language
+                        form.varicoseVeins
+                        (SetHealthEducationSubActivityBoolInput (\value form_ -> { form_ | varicoseVeins = Just value }))
+                        "varicose-veins"
+                        Nothing
+                  ]
+                , Just form.varicoseVeins
+                )
+
+            else
+                ( [], Nothing )
+
+        legPainRedness =
+            if provideLegPainRednessEducation assembled then
+                ( [ viewCustomLabel language (Translate.PrenatalHealthEducationLabel EducationLegPainRedness) "" "label header"
```

Comment:
It looks like this part of the code can put in a helper function with variables as it's repeated 6 times.

If possible, please export this code to a helper function.

```
if [condition here] then
  ( [ viewCustomLabel language (Translate.PrenatalHealthEducationLabel [variable]) "" "label header"
  , viewCustomLabel language Translate.PrenatalHealthEducationConstipationInform "." "label paragraph"
  , viewQuestionLabel language Translate.PrenatalHealthEducationAppropriateProvided
  , viewBoolInput
  language
  [variable]
  (SetHealthEducationSubActivityBoolInput (\value form_ -> { form_ | [variable] = Just value }))
  [variable]
  Nothing
  ]
  , Just [variable]
  )

else
  ( [], Nothing )
```

Something like that, sorry if I missed any variable.

Reply:
@nedSaf 
Not all that we have are the same, and it's not only variables, but set functions as well.
Since we're dealing with View here, I'd rather leave as is to prevent more complexity and have better readability.